providers/base/fetch.go

```diff
@@ -110,8 +110,37 @@ func (p *Provider[K, V]) startWebSocket(ctx context.Context, responseCh chan<- p
 			return ctx.Err()
 		default:
 			p.logger.Debug("starting websocket query handler")
-			if err := p.ws.Start(ctx, p.ids, responseCh); err != nil {
-				p.logger.Error("websocket query handler returned error", zap.Error(err))
+			// create sub handlers
```

Comment:
note that `p.ws.Start` will block the current thread/process - so this won't actually spin up multiple web socket connections. imo i think it makes more sense to first make the slices and then call `startWebSocket` with each slice in seperate go routines. `startWebSocket` should just function as it currently does but only has to deal with a subset of IDs.